modules/MicroMegaWattBSP/umwc_comms/evSerial.cpp

```diff
@@ -0,0 +1,417 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2021 Pionix GmbH and Contributors to EVerest
+#include "evSerial.h"
+#include "common.pb.h"
+#include "hi2lo.pb.h"
+#include "lo2hi.pb.h"
+#include "pb_decode.h"
+#include "pb_encode.h"
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include <errno.h>
+#include <fcntl.h>
+#include <string.h>
+
+#include <unistd.h>
+
+#include <thread>
+
+#include <date/date.h>
+#include <date/tz.h>
+
+evSerial::evSerial() {
+    fd = 0;
+    baud = 0;
+    reset_done_flag = false;
+    forced_reset = false;
+    cobsDecodeReset();
+}
+
+evSerial::~evSerial() {
+    if (fd)
+        close(fd);
+}
+
+bool evSerial::openDevice(const char* device, int _baud) {
```

Comment:
again this `openDevice` function, it might be preferable to even share large parts of the `evSerial.cpp` between Yeti and uMWC